### ISSUE TITLE
[Usage]: How to access to the generated token from LogitsProcessor

### ISSUE THREAD

**Issue by nguyen599** [vllm-project/vllm#10885]
### Your current environment

```text
The output of `python collect_env.py`
```

### How would you like to use vllm

Hi,

For some stupid reason, I want access to the current generated token from LogitsProcessor, something like:

```
tokenizer = llm.get_tokenizer()

class CustomLogitsProcessor:

    def __init__(self, tokenizer):
        self.tokenizer = tokenizer

    def __call__(
        self,
        generated_token,
        past_tokens_ids: Union[List[int], Tuple[int]],
        logits: torch.FloatTensor,
    ) -> torch.Tensor:
        current_generated_text = self.tokenizer.decode(generated_token)
        # do something ...

        return logits

```

Have any way to do it?

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by noooop** [comment]
It depends on whether you need to modify the logits and affect the "real" generated results.

If you need to modify the logits, you need to modify it in the sampler. 

Probably in [this place](https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/layers/sampler.py#L285C1-L293C10). before that，the token has not been generated yet.

If you don't need to modify logits, you can find it in [SamplerOutput](https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/layers/sampler.py#L97C1-L98C46). 


The easiest access is in [this place](
https://github.com/vllm-project/vllm/blob/b5b647b084de3a5a29d35ca527c9901f8e6a4e7e/vllm/engine/llm_engine.py#L1448C1-L1449C53)


Modifying such deep code is always hard, good luck

**Comment by nguyen599** [comment]
> I think you can't get generated_token from LogitsProcessor, it haven't sampled yet.
> 
> https://github.com/vllm-project/vllm/blob/b5b647b084de3a5a29d35ca527c9901f8e6a4e7e/vllm/worker/model_runner.py#L1725C1-L1738C10
> 
> It depends on whether you need to modify the logits and affect the "real" generated results.
> 
> If you need to modify the logits, you need to modify it in the sampler.
> 
> Probably in [this place](https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/layers/sampler.py#L285C1-L293C10). before that，the token has not been generated yet.
> 
> If you don't need to modify logits, you can find it in [SamplerOutput](https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/layers/sampler.py#L97C1-L98C46).
> 
> The easiest access is in [this place](https://github.com/vllm-project/vllm/blob/b5b647b084de3a5a29d35ca527c9901f8e6a4e7e/vllm/engine/llm_engine.py#L1448C1-L1449C53)
> 
> Modifying such deep code is always hard, good luck

@noooop thank you, I will try it.
